internal/btf/core.go

```diff
@@ -522,12 +571,31 @@ func adjustOffset(base uint32, t Type, n int) (uint32, error) {
 	return base + (uint32(n) * uint32(size) * 8), nil
 }
 
+// calculateBitfieldLoad computes the size and offset for loading a bitfield from
+// the target structure.
+func calculateBitfieldLoad(bitOffset, bitSize int) (size int, offset int, err error) {
+	// Start with the smallest possible aligned load before the bit offset.
+	size = 1
```

Comment:
Yep, this isn't doing the same as libbpf and it's not the "simpler" version you had. This version will use the smallest size load possible, so for this example it would do the following:
```
size=1, offset=8 / 8 / 1 * 1 = 1; 8+9 > (1+1*8) => 17 > 8 => load size too small
size=2, offset=8 / 8 / 2 * 2 = 0; 8+9 > (0+2*8) => 17 > 16 => load size too small
size=4, offset=8 / 8 / 4 * 4 = 0; 8+9 > (0+4*8) => 17 > 32 => load size big enough to capture the field
```

The calculation always produces an aligned offset:
```
offset = bitOffset / 8  # offset in bytes, rounded down, e.g. aligned to 1 byte.
offset = offset / size  # "number" of "size" blocks to reach the bit offset
offset = offset * size  # the offset aligned to "size" that is "just" before the bitOffset.
```

The benefit of this should be that we wouldn't read past the end of the struct if the bitfield was at the
end and was made smaller: `u64 foo:32, bar:32;` => `u32 bar:32`. We'd want a 4 byte load, but the
original load is 8 bytes and will read past the end.

Does that make sense or would you prefer that we'll use the libbpf version or your "simpler" version?
I'm fairly sure this should be an acceptable implementation for this, but please double-check :D

Reply:
Ok, it's failing when asked to compute the load for 31 bytes from bit offset 34 (bits 34..65), e.g. it'll try 1 byte from 32, then 2 bytes from 32, then 4 bytes from 32, and then 8 bytes from 0... and no solution. Should've written the exhaustive tests first :)

I'm on vacation now until Tuesday. I'm fix this then.

Reply:
OK, I'll take a closer look at your version as well. One thing to note before I forget: I think that using bitSize to derive alignment needs some documentation. This is because for something like `int b : 8` we'd have alignment of 4 but the bitSize would imply alignment of 1.

Regardless, we should document the following constraints:

* Must not read past the end of the struct
* Must produce a read <= bytes
* Must produce a read that is aligned